policy-controller/k8s/index/src/routes/grpc.rs

```diff
@@ -0,0 +1,22 @@
+use anyhow::Result;
+use linkerd_policy_controller_core::routes;
+use linkerd_policy_controller_k8s_api::gateway as k8s_gateway_api;
+
+pub fn try_match(
+    k8s_gateway_api::GrpcRouteMatch { headers, method }: k8s_gateway_api::GrpcRouteMatch,
+) -> Result<routes::GrpcRouteMatch> {
+    let headers = headers
+        .into_iter()
+        .flatten()
+        .map(super::http::header_match)
+        .collect::<Result<_>>()?;
```

Comment:
quick question about the behaviour here. iirc the route expects _all_ headers to match when specified so it would never make sense to ignore an error here, right?

e.g. say we have a list of three headers to match on. We can only fail here if we fail to parse any of the headers, for whatever reason. Failing to parse a header should fail the entire operation since we need all of them to match, we can't just ignore the "wrong" header.